thirdparty/runtime_environments.go

```diff
@@ -347,6 +365,98 @@ func stringToTime(timeInitial string) (time.Time, error) {
 	return time.Unix(timestamp, 0), nil
 }
 
+// Image stores information about an image including its name, version_id, kernel, ami, and its last deployed time.
+type Image struct {
+	Name         string
+	VersionID    string
+	Kernel       string
+	LastDeployed time.Time
+	AMI          string
+}
+
+// getNameFromOSInfo uses the provided AMI and name (exact match) arguments to filter the image information.
+func (c *RuntimeEnvironmentsClient) getNameFromOSInfo(ctx context.Context, ami string, name string) (string, error) {
+	opts := OSInfoFilterOptions{
+		AMI:  ami,
+		Name: fmt.Sprintf("^%s$", name),
+	}
+	result, err := c.GetOSInfo(ctx, opts)
+	if err != nil {
+		return "", errors.Wrap(err, "getting OS info")
+	}
+	if len(result) == 0 {
+		return "", errors.Errorf("OS information name '%s' not found for distro", opts.Name)
+	} else if len(result) > 1 {
+		return "", errors.Errorf("multiple results found for OS information name '%s'", opts.Name)
+	}
+	return result[0].Version, nil
+}
+
+// getLatestImageHistory returns the latest AMI and timestamp given the provided imageId.
+func (c *RuntimeEnvironmentsClient) getLatestImageHistory(ctx context.Context, imageID string) (string, time.Time, error) {
```

Comment:
I think having this return `(*ImageHistoryInfo, error)` might simplify some things (you can just return `nil`s)

You can probably do the timestamp conversion in the `GetImageInfo` function to accommodate for that